new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java

```diff
@@ -613,6 +614,98 @@ public void testDomainIntrospectVersionRolling() {
 
   }
 
+  /**
+   * Test, a new cluster can be created using online WLST change and new server pods can be brought
+   * up by patching domain resource with new cluster info and running the introspector.
+   */
+  @Order(3)
+  @Test
+  @DisplayName("Test new cluster creation on demand using WLST and introspection")
+  public void testCreateNewCluster() {
+
+    final String domainUid = "mydomain";
+    final String clusterName = "cl2";
+
+    final String adminServerName = "admin-server";
+    final String adminServerPodName = domainUid + "-" + adminServerName;
+
+    final String managedServerNameBase = "cl2-ms-";
+    String managedServerPodNamePrefix = domainUid + "-" + managedServerNameBase;
+
+    final int replicaCount = 2;
+
+    logger.info("Getting node port for default channel");
+    int adminServerT3Port = getServiceNodePort(introDomainNamespace, adminServerPodName + "-external", "t3channel");
+
+    // create a temporary WebLogic WLST property file
+    File wlstPropertiesFile = assertDoesNotThrow(() -> File.createTempFile("wlst", "properties"),
+        "Creating WLST properties file failed");
+    Properties p = new Properties();
+    p.setProperty("admin_host", K8S_NODEPORT_HOST);
+    p.setProperty("admin_port", Integer.toString(adminServerT3Port));
+    p.setProperty("admin_username", ADMIN_USERNAME_DEFAULT);
+    p.setProperty("admin_password", ADMIN_PASSWORD_DEFAULT);
+    p.setProperty("test_name", "create_cluster");
+    p.setProperty("cluster_name", clusterName);
+    p.setProperty("server_prefix", managedServerNameBase);
+    p.setProperty("server_count", "3");
+    assertDoesNotThrow(() -> p.store(new FileOutputStream(wlstPropertiesFile), "wlst properties file"),
+        "Failed to write the WLST properties to file");
+
+    // changet the admin server port to a different value to force pod restart
+    Path configScript = Paths.get(RESOURCE_DIR, "python-scripts", "introspect_version_script.py");
+    executeWLSTScript(configScript, wlstPropertiesFile.toPath(), introDomainNamespace);
+
+    logger.info("patch the domain resource with new cluster and introspectVersion");
+    String patchStr
+        = "["
+        + "{\"op\": \"add\",\"path\": \"/spec/clusters/-\", \"value\": "
+        + "    {\"clusterName\" : \"" + clusterName + "\", \"replicas\": 2, \"serverStartState\": \"RUNNING\"}"
+        + "},"
+        + "{\"op\": \"add\", \"path\": \"/spec/introspectVersion\", \"value\": \"4\"}"
```

Comment:
Can we get the current  Introspect version  and bump it instead of hard-coding to 4

Reply:
Good point. Added a new utility method getNextIntrospectVersion to auto increment the introspectVersion.